modules/unsupported/csv/src/main/java/org/geotools/data/csv/parse/CSVLatLonStrategy.java

```diff
@@ -65,26 +76,93 @@ protected SimpleFeatureType buildFeatureType() {
             Class<?> type = typesFromData.get(col);
             if (isLatitude(col)) {
                 latSpelling = col;
-                if (CSVStrategySupport.isNumeric(type)) {
+                if (CSVStrategy.isNumeric(type)) {
                     validLat = true;
                 }
             } else if (isLongitude(col)) {
                 lonSpelling = col;
-                if (CSVStrategySupport.isNumeric(type)) {
+                if (CSVStrategy.isNumeric(type)) {
                     validLon = true;
                 }
             }
         }
         if (validLat && validLon) {
-            builder.add(GEOMETRY_COLUMN, Point.class);
+        	List<String> csvHeaders = Arrays.asList(headers);
+        	int index = csvHeaders.indexOf(latSpelling);
+        	AttributeTypeBuilder builder2 = new AttributeTypeBuilder();
+        	builder2.setCRS(DefaultGeographicCRS.WGS84);
+        	builder2.binding(Point.class);       	
+        	AttributeDescriptor descriptor = builder2.buildDescriptor(GEOMETRY_COLUMN);
+        	builder.add(index, descriptor);
+			
             builder.remove(latSpelling);
             builder.remove(lonSpelling);
         }
         return builder.buildFeatureType();
     }
 
+    private boolean isLatitude(String s) {
+        return "latitude".equalsIgnoreCase(s) || "lat".equalsIgnoreCase(s);
+    }
+
+    private boolean isLongitude(String s) {
+        return "lon".equalsIgnoreCase(s) || "lng".equalsIgnoreCase(s) || "long".equalsIgnoreCase(s)
+                || "longitude".equalsIgnoreCase(s);
+    }
```

Comment:
I don't see a duplication? I think the confusion is because I moved the location of the methods - so it's showing them as a "diff" even though they aren't changed, just moved.